lib/eve.js

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect } from 'preact/hooks'
+
+// throttle a function call to avoid calling it too much
+export const throttle = (fn, delay, t, call = () => ((t = 0), fn())) => () =>
+  t || (t = setTimeout(call, delay))
+
+// create an observable or event
+export const eve = (...args) => {
+  const subs = new Set()
+  const sub = (fn) => (subs.add(fn), () => subs.delete(fn))
+  const next = () => new Promise(once)
+  const once = (fn) =>
+    subs.add(function $(value, prev) {
+      subs.delete($)
+      fn(value, prev)
+    })
+
+  // If no arguments are passed, this is an simple event
+  if (!args.length) {
```

Comment:
I use `args.length` here, this is a techniques that allows to know more precisely if the function was invoked with multiple arguments.

If I just check that the first element is `undefined`, then there is not difference between:
```js
Eve()  // create an event
```
and
```js
Eve(undefined) // observe from this value
```